internal/controller/driver_controller.go

```diff
@@ -1066,6 +1070,22 @@ func (r *driverReconcile) generateName(suffix string) string {
 	return fmt.Sprintf("%s-%s", r.driver.Name, suffix)
 }
 
+// generateServiceName generates a service name by replacing all special characters with a hyphen
+// as required by Kubernetes for service name.
+func (r *driverReconcile) generateServiceName(suffix string) (string, error) {
+	name := r.generateName(suffix)
+	// Define a regex pattern to match all special characters
+	re, err := regexp.Compile(`[^\w\s]`)
+	if err != nil {
+		return "", fmt.Errorf("failed to compile regex: %v", err)
+	}
```

Comment:
This seems wrong, you are selecting `\w` which in regexp selected all valid word chars

Reply:
All the validation already done on the name, to keep it simple just replacing all the special chars with `-`

Reply:
Something here is still not clear to me. Your regexp selects both `\w` and `\s` and then negates the selection. 
How does that fit into the category of all special chars?

Reply:
Let me make it simple and remove `\s` which seems to cause problem and driver should take care of it and replaces `\w` with `[^a-zA-Z0-9-]` where `/w` was allowing `_` which is also not valid

Reply:
@Madhu-1 
Checked the k8s docs. A service name must follow RFC 1035. This means the following:

- contain at most 63 characters
- contain only lowercase alphanumeric characters or '-'
- start with an alphabetic character
- end with an alphanumeric character

So lets add lowercasing of the name to your code